content/tokio/tutorial/async.md

```diff
@@ -557,18 +574,42 @@ impl Task {
         // Poll the future
         let _ = future.as_mut().poll(&mut cx);
     }
+
+    // Creates the Task given a Future and send it to execution
+    // for the very first time.
+    fn spawn<F>(future: F, sender: &channel::Sender<Arc<Task>>)
+    where
+        F: Future<Output = ()> + Send + 'static,
+    {
+        let task = Arc::new(Task {
+            future: Mutex::new(Box::pin(future)),
+            executor: sender.clone(),
+        });
+
+        let _ = sender.send(task);
+    }
+
 }
 ```
 
-Two things are happening here. First, `MiniTokio::run()` is implemented. The
-function runs in a loop receiving scheduled tasks from the channel. As tasks are
-pushed into the channel when they are woken, these tasks are able to make
-progress when executed.
+Multiple things are happening here. First, `MiniTokio::run()` is implemented.
+The function runs in a loop receiving scheduled tasks from the channel.
+As tasksare pushed into the channel when they are woken, these tasks are able
+to make progress when executed.
+
+Then the functions `MiniTokio::new()` and `MiniTokio::spawn()` are adjusted for
+working with the channel instead of the `VecDeque`. `MiniTokio::new()` creates
+the channel and sets it on the new `MiniTokio` before returning it.
+`MiniTokio::spawn()` receives the future and pass it to the `Task::spawn`
+function together with the send half of the channel..
 
 The `Task::poll()` function creates the waker using the [`ArcWake`] utility from
 the `futures` crate. The waker is used to create a `task::Context`. That
 `task::Context` is passed to `poll`.
 
+The `Task::spawn()` function receives a Future, creates a Task and send it to
+execution for the very first time, using its executor: send half of the channel.
+
```

Comment:
I know there are two `spawn` functions, but I think it's ok to just explain it once.

Reply:
I removed it based on your comment and on the fact that the comment from the [tutorial-code](https://github.com/tokio-rs/website/blob/48bff2dcceb4334b2cb1b8aeedf4dafe8c2134c8/tutorial-code/mini-tokio/src/main.rs#L68-L85) explains it too.